### DISCUSSION TITLE
How to use a Controller in another packages in workspace

### DISCUSSION THREAD

**Discussion by joeynq** [TypedProject/ts-express-decorators#2550] [Q&A] [answered]
Hi @Romakita ,

I have my Server.ts using a controller that exported from my other project and it returned 404.

When I copied the controller to the same package as `Server.ts`, it works.

Server.ts
```ts
// Not working
import { TheController } from "@repo/http"

@Configration({
  mount: {
    "/": [TheController]
  }
})
export class Server {}

// Working
@Controller("/")
class TheController {
  @Get("")
  get() { return "Hello world!" }
}

@Configration({
  mount: {
    "/": [TheController]
  }
})
export class Server {}
```

The same happened with Injectable service.

Do TsED support it or I'm doing something wrong?

**Comment by Romakita** [comment, marked as answer]
Hello @datnq 
You have problably something wrong in your side. Mono repo works.
https://github.com/tsedio/tsed-monorepo-template

I works always with mono repo for my customer project ^^.
tsed repo is also a mono repo.

The only problem could be a dependencies issue. All tsed dependencies must declared on package.json server level. Other packages must have tsed dependencies as peerDependencies. Otherwise, and depending on the package manager, it’ll install dependencies on each packages level.

See you

**Reply by joeynq** [reply]
Yes, it's on my side. I reinstalled my node_modules and it works again.

Thank you.

**Reply by joeynq** [reply]
Sorry to bother you again.

It seems working and the route was mounted, but it returned 404,

![carbon](https://github.com/tsedio/tsed/assets/2326129/b13388d1-fa2c-46fe-a85a-b7e9237b8151)

I'm using PNPM workspaces and can confirm all `@tsed/*` reference point to one `@tsed/*` instance.

In my `@repo/http`, tsed packages installed under peerDependencies
and `@repo/server`, tsed packages installed under dependecies

Also, I import `HomeController.ts` file from `@repo/http`, without transpiling

Edited:
It works if I use direct import `import { HomeController } from "@repo/http/src/controllers/HomeController";`

**Reply by Romakita** [reply]
Create repo example ;)

**Reply by joeynq** [reply]
Yes, It's definitely my mistake. I know you're right so I tried to start everything from scratch, carefully. Then I copied necessary codes to new project, it works. I still didn't know the root cause by good news is it's working now.

Appreciate your help. Thank you.

Updated: root cause is I configured Vite wrong, the root of vite was not point to expected path.
